### PR TITLE
Add nonlinear+linear splitting to ARK ODE solvers

### PR DESCRIPTION
**Pull request by mwarusz**

### CODE REVIEW

**Comment by mwarusz** [comment]
bors try

File: src/ODESolvers/AdditiveRungeKuttaMethod.jl [hunk 1]
```diff
@@ -327,6 +327,13 @@ ODEs.updatedt!(ark::AdditiveRungeKutta, dt) = ark.dt[1] = dt
 function ODEs.dostep!(Q, ark::AdditiveRungeKutta, param, timeend,
                       adjustfinalstep)
 
+  # this should be removed once we get rid of the old interface
+  if !isnothing(param)
```
Comment (jkozdon) [inline src/ODESolvers/AdditiveRungeKuttaMethod.jl]: Can we easily patch the old examples to remove this? (Just wondering)

**Comment by mwarusz** [comment]
bors try

File: src/ODESolvers/AdditiveRungeKuttaMethod.jl (hunk 1 above)
Comment (mwarusz) [reply src/ODESolvers/AdditiveRungeKuttaMethod.jl]: Good idea, thanks. Done.

**Comment by jkozdon** [comment]
Ready to merge?

**Comment by mwarusz** [comment]
Fine by me.

**Pull request merged**